clang/lib/CodeGen/Targets/NVPTX.cpp

```diff
@@ -203,8 +203,15 @@ ABIArgInfo NVPTXABIInfo::classifyArgumentType(QualType Ty) const {
 void NVPTXABIInfo::computeInfo(CGFunctionInfo &FI) const {
   if (!getCXXABI().classifyReturnType(FI))
     FI.getReturnInfo() = classifyReturnType(FI.getReturnType());
-  for (auto &I : FI.arguments())
-    I.info = classifyArgumentType(I.type);
+
+  unsigned ArgumentsCount = 0;
+  for (auto &I : FI.arguments()) {
+    if (FI.isVariadic() && ArgumentsCount > 0)
```

Comment:
Special-casing argument 0 is sort of unusual; are you sure this is what you want?  Normally, I'd expect the distinguishing factor to be fixed vs. variadic: so all explicitly specified parameters are treated the same, then all arguments in the variadic list are passed the same.

Reply:
You're right, this needs to account for all fixed arguments, not just the first (guaranteed) one. NVIDIA seems to handle it where the fixed arguments are passed using the regular ABI (can be indirect or direct) while the variadic arguments are always direct. Is there an easy way to check if an argument is part of the variadic set? Maybe if the argument number > number of arguments?

Reply:
Look at amdgpu, it does exactly that normal lowering for fixed and something different for variadic.